packages/keyring-controller/src/KeyringController.ts

```diff
@@ -2267,15 +2357,31 @@ export class KeyringController extends BaseController<
         assertIsExportableKeyEncryptor(this.#encryptor);
 
         if (password) {
-          const result = await this.#encryptor.decryptWithDetail(
-            password,
-            encryptedVault,
-          );
-          vault = result.vault;
-          this.#password = password;
+          if (encryptedEncryptionKey) {
```

Comment:
I feel like we are adding an important implementation detail (or an additional operation mode even) without properly documenting somewhere. Thoughts on adding a method like `#isEnvelopeModeEnabled()` to encode some documentation on this controller functionality?

Reply:
this is already documented at any public facing function, right?
though we can add more documentation, sure.
i like adding an explicit function for that check 👍 will do that